compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/model/types/BIterableTypeVisitor.java

```diff
@@ -109,14 +109,34 @@ public BIterableTypeVisitor(BLangDiagnosticLog dlog, SymbolTable symTable) {
 
     /* Util functions */
 
-    protected void logTooMayVariablesError(Operation op) {
+    protected void logTooManyVariablesError(Operation op) {
         dlog.error(op.pos, DiagnosticCode.ITERABLE_TOO_MANY_VARIABLES, op.collectionType);
     }
 
     protected void logNotEnoughVariablesError(Operation op, int count) {
         dlog.error(op.pos, DiagnosticCode.ITERABLE_NOT_ENOUGH_VARIABLES, op.collectionType, count);
     }
 
+    protected BType inferRecordFieldType(BRecordType recordType) {
```

Comment:
This method seems to be a duplicate of https://github.com/ballerina-platform/ballerina-lang/pull/10324/files#diff-782094a79e6831ac57a34ad157af49c1R490.

Shall we reuse that?